### DISCUSSION TITLE
[Support] 手动增加 bull 消费者报 undefined

### DISCUSSION THREAD

**Discussion by heathcliff-hu** [midwayjs/midway#4207] [Q&A] [answered]
## 问题

```
2024-12-01 00:22:57.026 ERROR 81177 [10 QueueTask] TypeError: Cannot read properties of undefined (reading 'name')
    at MidwayContainer.getAsync (/work/midway/packages/core/src/context/container.ts:604:45)
    at MidwayRequestContainer.getAsync (/work/midway/packages/core/src/context/requestContainer.ts:88:26)
    at BullQueue.<anonymous> (/work/midway/packages/bull/src/framework.ts:164:50)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
```

代码见 [diff](https://github.com/midwayjs/midway/compare/main...heathcliff-hu:midway:fix/undefined?expand=1)

执行该测试用例即可复现问题

```bash
npx lerna run test --scope=@midwayjs/bull
```

### Midway Versions（Midway 版本）

Need to install the following packages:
midway-version@1.4.0
Ok to proceed? (y) y

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(3.10.13)
✓ @midwayjs/fc-starter(3.19.0)
✓ @midwayjs/serverless-http-parser(3.14.0)
✓ @midwayjs/async-hooks-context-manager(3.19.0)
✓ @midwayjs/axios(3.19.1)
✓ @midwayjs/bootstrap(3.19.0)
✓ @midwayjs/bull(3.19.1)
✓ @midwayjs/bull-board(3.19.1)
✓ @midwayjs/busboy(3.19.0)
✓ @midwayjs/cache-manager(3.19.0)
✓ @midwayjs/captcha(3.19.0)
✓ @midwayjs/casbin(3.19.0)
✓ @midwayjs/casbin-redis-adapter(3.19.0)
✓ @midwayjs/casbin-typeorm-adapter(3.19.0)
✓ @midwayjs/code-dye(3.19.0)
✓ @midwayjs/consul(3.19.1)
✓ @midwayjs/core(3.19.0)
✓ @midwayjs/cos(3.19.0)
✓ @midwayjs/cron(3.19.0)
✓ @midwayjs/cross-domain(3.19.0)
✓ @midwayjs/decorator(3.19.0)
✓ @midwayjs/etcd(3.19.0)
✓ @midwayjs/express-session(3.19.0)
✓ @midwayjs/faas(3.19.0)
✓ @midwayjs/grpc(3.19.0)
✓ @midwayjs/http-proxy(3.19.0)
✓ @midwayjs/i18n(3.19.0)
✓ @midwayjs/info(3.19.0)
✓ @midwayjs/jwt(3.19.0)
✓ @midwayjs/kafka(3.19.0)
✓ @midwayjs/leoric(3.19.1)
✓ @midwayjs/mikro(3.19.1)
✓ @midwayjs/mock(3.19.0)
✓ @midwayjs/mongoose(3.19.1)
✓ @midwayjs/mqtt(3.19.1)
✓ @midwayjs/oss(3.19.0)
✓ @midwayjs/otel(3.19.1)
✓ @midwayjs/passport(3.19.0)
✓ @midwayjs/process-agent(3.19.0)
✓ @midwayjs/prometheus(3.19.0)
✓ @midwayjs/prometheus-socket-io(3.19.0)
✓ @midwayjs/rabbitmq(3.19.0)
✓ @midwayjs/redis(3.19.0)
✓ @midwayjs/security(3.19.0)
✓ @midwayjs/sequelize(3.19.0)
✓ @midwayjs/session(3.19.0)
✓ @midwayjs/socketio(3.19.0)
✓ @midwayjs/static-file(3.19.0)
✓ @midwayjs/swagger(3.19.0)
✓ @midwayjs/tablestore(3.19.0)
✓ @midwayjs/tags(3.19.0)
✓ @midwayjs/tenant(3.19.0)
✓ @midwayjs/typegoose(3.19.1)
✓ @midwayjs/typeorm(3.19.0)
✓ @midwayjs/upload(3.19.0)
✓ @midwayjs/validate(3.19.0)
✓ @midwayjs/version(3.19.1)
✓ @midwayjs/view(3.19.0)
✓ @midwayjs/view-ejs(3.19.0)
✓ @midwayjs/view-nunjucks(3.19.0)
✓ @midwayjs/web(3.19.0)
✓ @midwayjs/express(3.19.0)
✓ @midwayjs/koa(3.19.0)
✓ @midwayjs/ws(3.19.0)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment, marked as answer]
哦，因为不在 ioc 容器里了，加一个 provide 就行。
```ts
@Provide()
export class xxx {}
```

**Reply by heathcliff-hu** [reply]
多谢，加了之后没问题了。
